solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -311,7 +359,7 @@ boolean fetchLatestIndex(boolean forceReplication, boolean forceCoreReload) thro
         Replica replica = getLeaderReplica();
         CloudDescriptor cd = solrCore.getCoreDescriptor().getCloudDescriptor();
         if (cd.getCoreNodeName().equals(replica.getName())) {
-          return false;
+          return IndexFetchResult.CORE_NODE_IS_REPLICA;
```

Comment:
Maybe call this NOT_LEADER ?  Even a leader is a replica (it's the _role_ a replica plays).  Granted many of us promote this ambiguity because we haven't clearly/consistently labelled a non-leader, e.g. a follower.